app/models/transfer_request.rb

```diff
@@ -59,12 +59,20 @@ class TransferRequest < SystemRequest
     end
   end
 
-  before_create(:add_request_type)
+  before_validation :add_request_type, :set_request_purpose
+
+  has_many :transfer_request_collection_transfer_requests
+  has_many :transfer_request_collections, through: :transfer_request_collection_transfer_requests
+
   def add_request_type
```

Comment:
Create a concern to handle dynamic defaults and standardise the way defaults are created.